torch/_dynamo/variables/functions.py

```diff
@@ -689,31 +689,59 @@ def __init__(self, kernel, kernel_idx, grid, **kwargs):
     ) -> "VariableTracker":
+        from triton.runtime.autotuner import Autotuner
+
+        from .constant import ConstantVariable
         from .dicts import ConstDictVariable
         from .lists import BaseListVariable
 
-        grid = self.grid
-
-        if grid is None:
+        if self.grid is None:
             raise Unsupported("Triton kernels should always be called with a grid")
 
         # Both for grid's meta as well as for the kernel, we need combined
         # args and kwargs normalized
         normalized_args = {**dict(zip(self.kernel.arg_names, args)), **kwargs}
-        meta = ConstDictVariable(normalized_args, dict)
-
-        # If the grid is a function, then lets execute it and convert it to
-        # a list
-        if isinstance(grid, (NestedUserFunctionVariable, UserFunctionVariable)):
-            # Populate the special "meta" argument to call the grid function
-            grid = grid.call_function(tx, [meta], {})
 
-        # Now, the grid must be a list either originally or through above
-        # modification
-        if isinstance(grid, BaseListVariable):
-            grid = grid.as_proxy()
-        else:
-            unimplemented(f"grid for the triton kernel is {type(grid)}")
+        configs = (
+            [config.kwargs for config in self.kernel.configs]
+            if isinstance(self.kernel, Autotuner)
+            else [{}]
+        )
+        grids = []
+        for config_args in configs:
+            # If the grid is a function, then lets execute it and convert it to
+            # a list
+            grid = self.grid
+            if isinstance(grid, (NestedUserFunctionVariable, UserFunctionVariable)):
+                # Populate the special "meta" argument to call the grid function
+                config_args = {
+                    k: ConstantVariable.create(v) for k, v in config_args.items()
+                }
+                meta = ConstDictVariable({**normalized_args, **config_args}, dict)
+                grid = grid.call_function(tx, [meta], {})
+
+            # Now, the grid must be a list either originally or through above
+            # modification
+            if isinstance(grid, BaseListVariable):
+                grids.append(grid.as_proxy())
+            else:
+                unimplemented(f"grid for the triton kernel is {type(grid)}")
+
+        for i in range(len(grids)):
+            if not isinstance(grids[i], tuple):
+                raise Unsupported("Only tuple grids are supported")
+            # inductor expects all grids to be 3-tuple so lets make it
+            if len(grids[i]) == 1:
+                grids[i] = (grids[i][0], 1, 1)
+            elif len(grids[i]) == 2:
+                grids[i] = (grids[i][0], grids[i][1], 1)
+            elif len(grids[i]) > 3:
+                raise Unsupported("Grid can have at most rank 3")
+
+        assert len(grids) != 0
+        if len(set(grids)) == 1:
```

Comment:
This test causes SymInt specialization when there is SymInt in the grids, as SymInts are not hashable ordinarily.

Reply:
Is there a better way to do this or should we kill this optimization?

Reply:
Lets just kill it since this optimization is probably overkill anyway